### PR TITLE
Remove deprecated logic and warnings

### PR DESCRIPTION
**Pull request by amyeroberts**
# What does this PR do?

Kills a bunch of deprecated code

### CODE REVIEW

File: src/transformers/models/cohere/modeling_cohere.py [hunk 1]
```diff
@@ -651,11 +650,6 @@ def forward(
                 (see `past_key_values`).
             past_key_value (`Tuple(torch.FloatTensor)`, *optional*): cached past key and value projection states
         """
-        if "padding_mask" in kwargs:
```
Comment (ArthurZucker) [inline src/transformers/models/cohere/modeling_cohere.py:654]: kwargs should also be removed here.

Comment (ArthurZucker) [reply src/transformers/models/cohere/modeling_cohere.py:654]: some other PR are doing this as well

File: src/transformers/models/falcon/modeling_falcon.py [hunk 2]
```diff
@@ -395,11 +394,6 @@ def forward(
         output_attentions: bool = False,
         **kwargs,
```
Comment (ArthurZucker) [inline src/transformers/models/falcon/modeling_falcon.py:395]:
```suggestion
```

File: src/transformers/models/falcon/modeling_falcon.py [hunk 3]
```diff
@@ -551,14 +545,6 @@ def forward(
...
```
Comment (ArthurZucker) [inline src/transformers/models/falcon/modeling_falcon.py:546]:
```suggestion
```

File: src/transformers/models/falcon/modeling_falcon.py [hunk 4]
```diff
@@ -789,11 +775,6 @@ def forward(
...
```
Comment (ArthurZucker) [inline src/transformers/models/falcon/modeling_falcon.py:776]:
```suggestion
```

File: src/transformers/models/gemma/modeling_gemma.py
Comment (ArthurZucker) [inline src/transformers/models/gemma/modeling_gemma.py:1]: same here kwargs needs to be removed

File: src/transformers/models/llama/modeling_llama.py [hunk 5]
```diff
@@ -112,22 +111,6 @@ def __init__(self, dim, max_position_embeddings=2048, base=10000, device=None, s
...
-        return self._sin_cached
-
-    @property
-    def cos_cached(self):
-        logger.warning_once(
-            "The cos_cached attribute will be removed in 4.39. Bear in mind that its contents changed in v4.38. Use "
-            "the forward method of RoPE from now on instead. It is not used in the `LlamaAttention` class"
-        )
-        return self._cos_cached
-
```
Comment (ArthurZucker) [inline src/transformers/models/llama/modeling_llama.py:130]: the buffers that are registered need to be deleted as well

File: src/transformers/models/llama/modeling_llama.py [hunk 6]
```diff
@@ -730,11 +713,6 @@ def forward(
                 (see `past_key_values`).
             past_key_value (`Tuple(torch.FloatTensor)`, *optional*): cached past key and value projection states
         """
-        if "padding_mask" in kwargs:
```
Comment (ArthurZucker) [inline src/transformers/models/llama/modeling_llama.py:733]: kwargs need removal

File: src/transformers/models/maskformer/modeling_maskformer.py [hunk 7]
```diff
@@ -441,22 +441,9 @@ def _shape(self, tensor: torch.Tensor, seq_len: int, batch_size: int):
         return tensor.view(batch_size, seq_len, self.num_heads, self.head_dim).transpose(1, 2).contiguous()
 
     def with_pos_embed(self, tensor: torch.Tensor, object_queries: Optional[Tensor], **kwargs):
-        position_embeddings = kwargs.pop("position_embeddings", None)
```
Comment (ArthurZucker) [inline src/transformers/models/maskformer/modeling_maskformer.py:444]: do we still need the kwargs?

File: src/transformers/models/mistral/modeling_mistral.py
Comment (ArthurZucker) [inline src/transformers/models/mistral/modeling_mistral.py:1]: same comments as for llama!

File: src/transformers/models/phi3/modeling_phi3.py
Comment (ArthurZucker) [inline src/transformers/models/phi3/modeling_phi3.py:1]: same comment as llama

**Review by ArthurZucker — COMMENTED** [review]
Thanks for the cleanup! A few nits here and there and we'll be good to go

**Comment by amyeroberts** [comment]
@ArthurZucker Thanks for the review! I've removed all of the kwargs being passed now, and the `_cos_cached`, `_sin_cached` buffers

**Review by ArthurZucker — APPROVED** [review]
Thanks 🤗

**Pull request merged**